client/keys/utils.go

```diff
@@ -21,6 +21,8 @@ const KeyDBName = "keys"
 // keybase is used to make GetKeyBase a singleton
 var keybase keys.Keybase
 
+// TODO make keybase take a database not load from the directory
```

Comment:
Let's move this into an issue, `TODO`s in the code are hard to track - https://github.com/cosmos/cosmos-sdk/issues/1205

Reply:
closing for now I don't think this is necessarily achievable without a serious refactor to keystore / not using viper - added an additional comment but I think that not using viper should cover this issue